src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/CqlProxyRetryPolicy.java

```diff
@@ -0,0 +1,136 @@
+package io.stargate.sgv2.jsonapi.service.cqldriver;
+
+import static com.datastax.oss.driver.internal.core.retry.DefaultRetryPolicy.RETRYING_ON_UNAVAILABLE;
+
+import com.datastax.oss.driver.api.core.ConsistencyLevel;
+import com.datastax.oss.driver.api.core.context.DriverContext;
+import com.datastax.oss.driver.api.core.retry.RetryDecision;
+import com.datastax.oss.driver.api.core.retry.RetryPolicy;
+import com.datastax.oss.driver.api.core.retry.RetryVerdict;
+import com.datastax.oss.driver.api.core.servererrors.CoordinatorException;
+import com.datastax.oss.driver.api.core.servererrors.WriteType;
+import com.datastax.oss.driver.api.core.session.Request;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This retry policy while executing cql statements is useful when the requests are delegated to a
+ * proxy layer which is responsible for retrying the requests. This policy will only retry once if
```

Comment:
But weren't we thinking that we do NOT want proxy layer doing retries, since we anyway require non-proxy retries & try to avoid "double retry"?
So is there need for this policy?

Reply:
No, we would to have the proxy layer do the retry and have the JSON API driver retry only when the proxy layer is unavailable.

Reply:
Ah ok. So whether this policy is used is separate from it being available for use. Makes sense.